gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -1976,6 +1982,12 @@ gst_tensor_filter_common_set_property (GstTensorFilterPrivate * priv,
     case PROP_SHARED_TENSOR_FILTER_KEY:
       status = _gtfc_setprop_SHARED_TENSOR_FILTER_KEY (prop, value);
       break;
+    case PROP_LATENCY_REPORT:
+      priv->latency_reporting = g_value_get_boolean (value);
+      /* latency reporting requires latency_mode=1 to compute latency estimates */
```

Comment:
1. If it is required, the property description should let users know.

2. I'd suggest to mention that latency_report will be only effective if latency_mode is 1 and ignore latency_report if it is kept 0 in run-time. You may emit a warning message for such inconsistency.

If latency_mode=1 is set forcibly when latency_reporting=1, some frameworks/applications on the of gstreamer that do set-property repeatedly for a given element (e.g., pipeviz) may suffer from undesired behaviors.

Or you may update "need_profiling" accordingly and rely on it around LINE 788

Reply:
Agreed, it is clearer to keep properties and associated variables independent from each others.

Change updated to no longer couple latency_mode variable to the latency_report setting.
Logic has been moved to  need_profiling definition and record_statistics(), solely based on latency_reporting variable that reflects latency_report setting.